packages/starlight/schemas/i18n.ts

```diff
@@ -23,16 +23,26 @@ const defaultI18nSchema = () =>
 /** Type of Starlight’s default i18n schema, including extensions from Pagefind and Expressive Code. */
 type DefaultI18nSchema = ReturnType<typeof defaultI18nSchema>;
 
+/**
+ * Based on the the return type of Zod’s `merge()` method. Merges the type of two `z.object()` schemas.
+ * Also sets them as “passthrough” schemas as that’s how we use them. In practice whether or not the types
+ * are passthrough or not doesn’t matter too much.
```

Comment:
You mean something like this @HiDeoo?

```suggestion
 * are passthrough or not doesn’t matter too much.
 * 
 * @see https://github.com/colinhacks/zod/blob/3032e240a0c227692bb96eedf240ed493c53f54c/src/types.ts#L2656-L2660
```

Reply:
Exactly, the previous `z.ZodIntersection` was very explicit and the new one can be kinda scary or magical if you don't exactly know where it comes from imo. Your call.

Reply:
OK. Will add this. I also realised doing this that `ZodIntersection` was completely the wrong type technically — `merge()` doesn’t return an intersection but a composite object — but never mind, it worked anyway 😁